src/app/libp2p_helper/src/codanet.go

```diff
@@ -257,7 +282,7 @@ func (gs *CodaGatingState) InterceptSecured(_ network.Direction, id peer.ID, add
 	// connections in coda are symmetric: if i am allowed to connect to
 	// you, you are allowed to connect to me.
 	remoteAddr := addrs.RemoteMultiaddr()
-	allow = gs.AllowedPeers.Contains(id) || (!gs.DeniedPeers.Contains(id) && !gs.AddrFilters.AddrBlocked(remoteAddr))
+	allow = gs.isAllowedPeer(id)
```

Comment:
can you add gs.isAllowedAddr here as well? Maybe the state machine will work for us, but just in case. Keep AllowedPeers.Contains in front too.